### ISSUE TITLE
[Crash] Media instance not initialized

### ISSUE THREAD

**Issue by Symbai** [ZeBobo5/Vlc.DotNet#292]
I call duration via timer like this:

`VlcMedia1.GetCurrentMedia?.Duration`

after some time the application crashes and I receive an error saying "Media instance not initliazed".
Stacktrace;
```

 at Vlc.DotNet.Core.Interops.VlcManager.GetMediaDuration(Vlc.DotNet.Core.Interops.VlcMediaInstance)
   at Vlc.DotNet.Core.VlcMedia.get_Duration()
   at VideoCenter.Form1.ProgressBarControl1_CustomDisplayText(System.Object, DevExpress.XtraEditors.Controls.CustomDisplayTextEventArgs)
   at VideoCenter.Form1.TimerUpdatePos_Tick(System.Object, System.EventArgs)
   at System.Windows.Forms.Timer.OnTick(System.EventArgs)
   at System.Windows.Forms.Timer+TimerNativeWindow.WndProc(System.Windows.Forms.Message ByRef)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr, Int32, IntPtr, IntPtr)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG ByRef)
   at System.Windows.Forms.Application+ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr, Int32, Int32)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoopInner(Int32, System.Windows.Forms.ApplicationContext)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoop(Int32, System.Windows.Forms.ApplicationContext)
   at System.Windows.Forms.Application.RunDialog(System.Windows.Forms.Form)
   at System.Windows.Forms.Form.ShowDialog(System.Windows.Forms.IWin32Window)
   at System.Windows.Forms.Form.ShowDialog()
   at System.Windows.Forms.Application+ThreadContext.OnThreadException(System.Exception)
   at System.Windows.Forms.Timer+TimerNativeWindow.OnThreadException(System.Exception)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr, Int32, IntPtr, IntPtr)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG ByRef)
   at System.Windows.Forms.Application+ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr, Int32, Int32)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoopInner(Int32, System.Windows.Forms.ApplicationContext)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoop(Int32, System.Windows.Forms.ApplicationContext)
   at Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase.OnRun()
   at Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase.DoApplicationModel()
   at Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase.Run(System.String[])
   at VideoCenter.My.MyApplication.Main(System.String[])
```

**Comment by jeremyVignelles** [comment]
Could not reproduce.
Which version of libvlc are you using? Could you provide a sample project so that we can give it a try?

**Comment by Symbai** [comment]
It happens sporadic so there is no code that lets you reproduce it 100% of the time.
Here is my code. When the error occurs VS breaks at the first line where I call the duration.

```
Private Sub ProgressBarControl1_CustomDisplayText(sender As Object, e As CustomDisplayTextEventArgs) Handles ProgressBarControl1.CustomDisplayText
               Dim dur = VlcControl1.GetCurrentMedia?.Duration
        If dur Is Nothing Then Return


        e.DisplayText = TimeSpan.FromSeconds(dur?.TotalSeconds * VlcControl1.Position).ToString("hh\:mm\:ss") & " /  " & dur?.Duration.ToString("hh\:mm\:ss")
        ProgressBarControl1.Update()
    End Sub
```

**Comment by jeremyVignelles** [comment]
Do you have multiple players in your app? Does it happen when the video plays or at the beginning/end of the video?

**Comment by Symbai** [comment]
No I just have one. It happens at playback. For example the video plays for 10 minutes in a row without an issue, then in the middle of the video suddently I receive the exception report.

But I have to mention that in my app I skip through several videos (only once is playing at the time though). So perhaps the issue is kind of left over / async issue?
I haven't tested with just playing a single video file without stopping and playing another video.

**Comment by pimpmypc89** [comment]
If you check myIsDisposing when it crashes, you'll probably find it is set to True.

**Comment by Symbai** [comment]
Well but what does that mean? Does it mean I did something wrong (which would it be?) and how would I fix it in code?

**Comment by pimpmypc89** [comment]
Someone else might want to chime in, but its a bug I also experienced, and still haven't tracked down. The vlcMediaInstance gets disposed and then it breaks.

**Comment by jeremyVignelles** [comment]
Do you have any `Dispose` getting called in your app? or maybe are you using a `using` somewhere that could dispose that instance?

You could try to get the source code of this project, copy the projects in your solution and use those rather than using the NuGet package. Then, you could set a breakpoint in the `Dispose` method of the Media player and the Media, and see when is that getting called?

**Comment by Symbai** [comment]
No dipose and no using (except of  openfiledialog once). The only thing that might be important enough is that I use a fix discussed here at github in a different ticket to make a video loop using the following code:

```
    Private Delegate Sub delegate_vlc_play(file As Uri, pars As String())
    Private Sub VlcControl1_EndReached(sender As Object, e As VlcMediaPlayerEndReachedEventArgs) Handles VlcControl1.EndReached
        Dim del_vlc As New delegate_vlc_play(AddressOf VlcControl1.Play)
        del_vlc.BeginInvoke(New Uri(VlcControl1.GetCurrentMedia?.Mrl), New String() {}, Nothing, Nothing)
    End Sub
```

**Comment by dilandau2001** [comment]
This might be also fixed with this pull #477 
Calling GetCurrentMedia was creating a new VlcMedia everytime.
If you did it in a regular basis then you would be having a memory leak with lots of initialized VlcMedias for a single player.

Then, those "forgotten" VlcMedias were released somehow at a not controlled moment which was equivalent to call dispose.

I was able to reproduce this problem many times during the development of this pull request so I feel this issue might be fixed also.
